app/src/components/page/Sidebar/index.tsx

```diff
@@ -58,13 +25,41 @@ const defaultSidebarProps: Partial<SidebarProps> = {
 
 /** The Sidebar component */
 export const SidebarComponent: React.FC<SidebarProps> = (props: SidebarProps) => {
+  useTranslation();
   const { extraData } = props;
   const { roles } = extraData;
-  const activeRoles = OPENSRP_ROLES;
   let location = useLocation();
   let loc = location.pathname.split('/');
-  loc.shift();
-  useTranslation();
+  const [openKeys, setOpenKeys] = React.useState<React.Key[]>([]);
+
+  const routes = React.useMemo(() => getRoutes(roles as string[]), [roles]);
+
+  const mapChildren = React.useCallback((route: Route) => {
+    if (route.children) {
+      return (
+        <Menu.SubMenu key={route.key} icon={route.otherProps?.icon} title={route.title}>
+          {route.children.map(mapChildren)}
+        </Menu.SubMenu>
+      );
+    } else if (route.url) {
+      return (
+        <Menu.Item key={route.key}>
+          <Link className="admin-link" to={route.url}>
+            {route.title}
+          </Link>
+        </Menu.Item>
+      );
+    } else {
+      return <Menu.Item key={route.key}>{route.title}</Menu.Item>;
+    }
+  }, []);
+
+  const sidebaritems: JSX.Element[] = React.useMemo(() => {
+    return routes.filter((route) => route.enabled).map(mapChildren);
+  }, [mapChildren, routes]);
```

Comment:
mapChildren is only used here, can be declared inside here and avoid passing it as a dependency or memoizing it.

Reply:
not sure if it be nicer in term of memorizations or cache 🤔💭 . will it be ?

Reply:
The function itself is not memoized as it is (It has nothing in it's dependency array). See this https://github.com/OpenSRP/web/pull/395#discussion_r602419306

Reply:
Done